### PR TITLE
Enable query by configuration variable id in order to resolve #215

### LINKED ISSUE #215: Terraform Provider - Overriding a var with is_read_only corrupts the state
1. Add the following resources:

```
resource "env0_configuration_variable" "orgvar" {
  name = "myvar"
  value = "orgvalue"
  is_read_only = true
}

resource "env0_configuration_variable" "projvar" {
  name = env0_configuration_variable.orgvar.name
  value = "projvar"
  project_id = "<some_project_id>"
}
```

2. Apply the TF file. That was allowed
3. Going to env0, you don't see the project variable, but it was created
4. Try terraform apply again, you get an error. You now can't even remove the resource from terraform. The only thing you can do is manually remove the variables from env0

```
Error: variable 8d7d8d31-7600-409f-af3c-269ca09c5fdf not found (under this scope): <nil>
│
│   with env0_configuration_variable.projvar,
│   on main.tf line 24, in resource "env0_configuration_variable" "projvar":
│   24: resource "env0_configuration_variable" "projvar" {
```

### PR DESCRIPTION
**Pull request by eranelbaz**
### Issue & Steps to Reproduce / Feature Request
[//]: <> (choose “resolves” “fixes” or “closes” and put link for the issue)
fixes #215 
The problem is that when we query by scope with `readonly` we don't return the lower scopes

### Solution
added query by id function to our client
if we have id use it

also added test that check #215

also saw we are missing 
https://github.com/env0/terraform-provider-env0/blob/11466150976f86a7615b14116d8fa0c112613563/env0/resource_configuration_variable.go#L211-L212
in our read function

### CODE REVIEW

**Comment by eranelbaz** [comment]
[Terraform Provider - Overriding a var with is_read_only corrupts the state](https://trello.com/c/qXjSVCH9/5309-terraform-provider-overriding-a-var-with-isreadonly-corrupts-the-state)

File: client/api_client.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ type ApiClient struct {
 
 type ApiClientInterface interface {
 	ConfigurationVariables(scope Scope, scopeId string) ([]ConfigurationVariable, error)
```
Comment (yaronya) [inline client/api_client.go:15]: Maybe worth changing existing method to `ConfigurationVariablesByScope` ?
No biggie, your call

File: client/configuration_variable_test.go [hunk 2]
```diff
@@ -35,6 +35,24 @@ var _ = Describe("Configuration Variable", func() {
...
+			httpCall = mockHttpClient.EXPECT().
+				Get("/configuration/"+id, nil, gomock.Any()).
+				Do(func(path string, request interface{}, response *ConfigurationVariable) {
+					*response = mockConfigurationVariable
+				})
+
+			found, _ = apiClient.ConfigurationVariablesById(id)
+		})
+
+		It("Should return created triggers", func() {
```
Comment (yaronya) [inline client/configuration_variable_test.go:51]: Text is wrong

File: env0/resource_configuration_variable.go [hunk 3]
```diff
@@ -198,36 +198,34 @@ func resourceConfigurationVariableRead(ctx context.Context, d *schema.ResourceDa
...
 
-				if variable.Schema.Format != "" {
-					d.Set("format", variable.Schema.Format)
-				}
-			}
+	d.Set("name", variable.Name)
+	d.Set("description", variable.Description)
+	d.Set("value", variable.Value)
+	d.Set("is_sensitive", variable.IsSensitive)
+	d.Set("is_read_only", variable.IsReadonly)
```
Comment (yaronya) [inline env0/resource_configuration_variable.go:211]: This means that we haven't QA-ed drift detection properly for configuration properties 😞

**Review by yaronya — APPROVED** [review]
LGTM
Left some minor comments

File: client/configuration_variable_test.go (hunk 2 above)
Comment (eranelbaz) [reply client/configuration_variable_test.go]: 1fde85e7(rename2)

File: client/api_client.go (hunk 1 above)
Comment (eranelbaz) [reply client/api_client.go]: a66cb455(rename)

**Pull request merged**